argoverse/utils/ply_loader.py

```diff
@@ -26,3 +26,37 @@ def load_ply(ply_fpath: _PathLike) -> np.ndarray:
     z = np.array(data.points.z)[:, np.newaxis]
 
     return np.concatenate((x, y, z), axis=1)
+
+
+def load_ply_by_attrib(ply_fpath: _PathLike, attrib_spec: str) -> Optional[np.ndarray]:
+    """Load a point cloud file from a filepath.
+
+    Args:
+        ply_fpath: Path to a PLY file
+        attrib_spec: string of C characters, each char representing a desired point attribute
+            x -> point x-coord
+            y -> point y-coord
+            z -> point z-coord
+            i -> point intensity/reflectance
+            l -> laser number of laser from which point was returned
+
+    Returns:
+        arr: Array of shape (N, C). If attrib_str is invalid, `None` will be returned
+    """
+    possible_attributes = ["x", "y", "z", "i", "l"]
```

Comment:
I wonder if we should do this with an `OrderedDict`? Probably doesn't matter too much, but we could do something like:

```
possible_attributes = OrderedDict({"x": None, "y": None, "z": None, "i": None, "l": None})
filtered_attributes = list(filter(lambda x: x in possible_attributes, list(attrib_spec)))
if not filtered_attributes:
    return None

data = pyntcloud.PyntCloud.from_file(os.fspath(ply_fpath))
return np.stack([getattr(data.points, name) for name in filtered_attributes], axis=1)
```

Just a thought, though.

Reply:
Actually, I think `OrderedDict` and just be a `Dict`.